weave/trace/client_progress_bar.py

```diff
@@ -0,0 +1,122 @@
+"""Progress bar utilities for WeaveClient.
+
+This module provides functionality for displaying progress bars when flushing
+tasks in the WeaveClient.
+"""
+
+from typing import Callable, TypedDict
+
+from rich.console import Console
+from rich.progress import (
+    BarColumn,
+    Progress,
+    SpinnerColumn,
+    TaskProgressColumn,
+    TextColumn,
+    TimeElapsedColumn,
+)
+
+
+class WeaveFlushStatus(TypedDict):
```

Comment:
nit: i would move this type def to weave_client

Reply:
My inclination is to keep the client as clean as possible, but I agree it does sort of want to be there.